### PR TITLE
Account creation: Optimizations with separate fields and navigation to login if possible

### LINKED ISSUE #9407: Account creation: Separate email and password
frame-1000004590

Internal ref: pecCkj-tX-p2

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes #9407 

## Description

This PR adds optimization to the account creation flow following the latest updates in pecCkj-wI-p2:
- Separates fields email and password during account creation. Since the two screens are almost identical, I reuse the same view and view model, with the new parameter `field` to decide which field to show.
- Handles existing email error by navigating to the login flow. Please also review https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/767 to see the changes in the WordPressAuthenticator library.
- Changes some wording: main CTA button, the title of the password screen, and moves the Log In button to the navigation bar.
- Changes the title of the Reset Password button in the Login flow to Forgot your password?

## Testing instructions

- On the prologue screen tap Get Started.
- Enter the email address of an existing email and tap Continue. Notice that the login flow is presented with the input email already populated.
- Navigate back to the account creation screen and change the email to a new one. Tap Continue and notice another screen is displayed with the password field only.
- Enter a valid password and tap Continue. A new account should be created successfully.
- Please feel free to test with invalid input to see if the error cases are still handled correctly.

## Screenshots

https://user-images.githubusercontent.com/5533851/232018749-ea238227-ea15-470f-b87b-e5ea74129414.mp4

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by selanthiraiyan — COMMENTED** [review]
I noticed the following cases while testing the app (before I started reviewing the code),

1. Already existing email 
ExistingAccount

- Stays in the "Get started - Email" screen after loading indicator for a while. There is no error message. 

2. New account signup

https://user-images.githubusercontent.com/524475/232395925-c55a9904-06e1-4007-a914-241589c43da1.mp4

The password screen is presented as expected. But I see the following events being tracked.
```
🔵 Tracked signup_submitted, properties: [:]
🔵 Tracked signup_failed, properties: [AnyHashable("error_type"): "PASSWORD_INVALID"]
```
The signup_failed event is tracked even before I enter a password. 

Please correct me if I am missing something here.  I ensured that,
- I pulled the latest trunk.
- Deleted `Pods` folder and executed `bundle install && bundle exec pod install`

**Comment by itsmeichigo** [comment]
@selanthiraiyan Thanks for the reviews! I updated the navigation and tracking in f4e80a2b155a99f3d5056fc688b5386975a4ddf3 and 448ace94c6a0b810f9aa8a607f9a0171322bd8f5 - please take another look!

File: WooCommerce/WooCommerceTests/ViewModels/Authentication/AccountCreationFormViewModelTests.swift [hunk 1]
```diff
@@ -173,6 +222,32 @@ final class AccountCreationFormViewModelTests: XCTestCase {
...
+        viewModel.email = "test@example.com"
+
+        // When
+        try? await viewModel.createAccount()
+
+        // Then
+        XCTAssertEqual(analyticsProvider.receivedEvents, ["signup_submitted"])
+    }
+
+    func test_createAccount_failure_with_invalid_password_is__tracked_if_emailSubmissionHandler_is_not_available() async {
```
Comment (selanthiraiyan) [inline WooCommerce/WooCommerceTests/ViewModels/Authentication/AccountCreationFormViewModelTests.swift]: Typo
```suggestion
    func test_createAccount_failure_with_invalid_password_is_tracked_if_emailSubmissionHandler_is_not_available() async {
```

**Review by selanthiraiyan — APPROVED** [review]
Thanks for adding unit tests. 🎉 

This works as expected. 🚢

**Pull request merged**